_inc/client/rest-api/index.js

```diff
@@ -10,244 +10,198 @@ import assign from 'lodash/assign';
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce
-			}
-		} )
-		.then( checkStatus ).then( response => response.json() ),
-		fetchSettings: () => fetch( `${ apiRoot }jetpack/v4/settings`, {
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce
-			}
-		} )
-		.then( checkStatus ).then( response => response.json() ),
-		updateSetting: ( updatedSetting ) => fetch( `${ apiRoot }jetpack/v4/settings`, {
-			method: 'post',
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce,
-				'Content-type': 'application/json'
-			},
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		fetchStatsData: ( range ) => getRequest( statsDataUrl( range ) )
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		getPluginUpdates: () => getRequest( `${ apiRoot }jetpack/v4/updates/plugins` )
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		fetchSettings: () => getRequest( `${ apiRoot }jetpack/v4/settings` )
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		updateSetting: ( updatedSetting ) => postRequest( `${ apiRoot }jetpack/v4/settings`, {
 			body: JSON.stringify( updatedSetting )
 		} )
-		.then( checkStatus ).then( response => response.json() ),
-		fetchSiteData: () => {
-			return fetch( `${ apiRoot }jetpack/v4/site`, {
-				method: 'get',
-				credentials: 'same-origin',
-				headers: {
-					'X-WP-Nonce': apiNonce
-				}
-			} )
-			.then( checkStatus ).then( response => response.json() )
-			.then( body => {
-				return JSON.parse( body.data );
-			} );
-		},
-		dismissJetpackNotice: ( notice ) => fetch( `${ apiRoot }jetpack/v4/notice/${ notice }`, {
-			method: 'post',
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce,
-				'Content-type': 'application/json'
-			},
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		fetchSiteData: () => getRequest( `${ apiRoot }jetpack/v4/site` )
+			.then( checkStatus )
+			.then( response => response.json() )
+			.then( body => JSON.parse( body.data ) ),
+
+		dismissJetpackNotice: ( notice ) => postRequest( `${ apiRoot }jetpack/v4/notice/${ notice }`, {
 			body: JSON.stringify( { dismissed: true } )
 		} )
-		.then( checkStatus ).then( response => response.json() ),
-		fetchPluginsData: () => fetch( `${ apiRoot }jetpack/v4/plugins`, {
-			method: 'get',
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce
-			}
-		} )
-		.then( checkStatus ).then( response => response.json() )
+			.then( checkStatus )
+			.then( response => response.json() ),
+
+		fetchPluginsData: () => getRequest( `${ apiRoot }jetpack/v4/plugins` )
+			.then( checkStatus )
+			.then( response => response.json() )
 	};
 
+	function addCacheBuster( route ) {
+		const parts = route.split( '?' ),
+			query = parts.length > 1
+				? parts[ 1 ]
+				: '',
+			args = query.split( '&' );
+
+		args.push( '_rnd=' + ( 10e9 * Math.random() ) );
```

Comment:
instead of doing generating a random math character. I would suggest we just use the current time in seconds or milliseconds with 
```
var ddate = new Date();
var milliseconds = d.getTime();
```

also can we make the `_rnd` more explicit with something like `_cacheBuster`